src/dom/context2d.mjs

```diff
@@ -1,10 +1,15 @@
-export default function(width, height, dpi) {
-  if (dpi == null) dpi = devicePixelRatio;
-  var canvas = document.createElement("canvas");
-  canvas.width = width * dpi;
-  canvas.height = height * dpi;
+export default function (width, height, options = {}) {
+  const {scale = devicePixelRatio, ...contextOptions} = !isNaN(options)
+    ? {...(options != null && {scale: options})}
```

Comment:
There's no need for the destructuring here, and the null check should be explicit. If it is undefined, we already get `{}`.
```suggestion
    ? {scale: options} ?? {}
```